replication_test.go

```diff
@@ -173,6 +174,17 @@ func TestClientIMPL_GetReplicationSessionByLocalResourceID(t *testing.T) {
 	resp, err := C.GetReplicationSessionByLocalResourceID(context.Background(), volID2)
 	assert.Nil(t, err)
 	assert.Equal(t, volID, resp.ID)
+
+	// test when the replication group does not exist
+	httpmock.Reset()
+
+	httpmock.RegisterResponder("GET", replicationSessionMockURL,
+		httpmock.NewStringResponder(200, ""))
+	_, err = C.GetReplicationSessionByLocalResourceID(context.Background(), volID2)
+
+	assert.NotNil(t, err)
+	apiError := err.(APIError)
+	assert.True(t, apiError.NotFound())
 }
```

Comment:
It is bad style to have these two tests in the same test method. Typically only done when the setup is very expensive and you can quickly test two things in the same method. In those cases it is better to use a Context to pass between tests.

Reply:
Thanks for the info. It appears this is normal implementation for this library. I'll update the docs created to reflect this so we can avoid it in the future.

Reply:
Also note that testing two things in the same test counts as just one test metric (unless the framework does some magic). Potentially if the first test failed then the second one is blocked. If they were separate tests then potentially the second test could run and pass. So you increase your test failure rate for no good reason.